libbeat/processors/script/javascript/module/processor/processor.go

```diff
@@ -37,91 +36,101 @@ import (
 	"github.com/elastic/beats/libbeat/processors/script/javascript"
 )
 
-// newConstructor returns a JS constructor function. The constructor wraps a
-// beat processor constructor. The javascript constructor must be passed a value
-// that can be treated as the processor's config.
-func newConstructor(
-	s *goja.Runtime,
-	constructor processors.Constructor,
-) func(call goja.ConstructorCall) *goja.Object {
-	return func(call goja.ConstructorCall) *goja.Object {
-		a0 := call.Argument(0)
-		if a0 == nil {
-			panic(s.ToValue("constructor requires a configuration arg"))
-		}
-
-		commonConfig, err := common.NewConfigFrom(a0.Export())
-		if err != nil {
-			panic(s.NewGoError(err))
-		}
-
-		p, err := constructor(commonConfig)
-		if err != nil {
-			panic(s.NewGoError(err))
-		}
-
-		bp := &beatProcessor{s, p}
-		call.This.Set("Run", bp.Run)
-		return nil
-	}
+// Create constructors for most of the Beat processors.
+// Note that script is omitted to avoid nesting.
+var constructors = map[string]processors.Constructor{
+	"AddCloudMetadata":      add_cloud_metadata.New,
+	"AddDockerMetadata":     add_docker_metadata.New,
+	"AddFields":             actions.CreateAddFields,
+	"AddHostMetadata":       add_host_metadata.New,
+	"AddKubernetesMetadata": add_kubernetes_metadata.New,
+	"AddLocale":             add_locale.New,
+	"AddProcessMetadata":    add_process_metadata.New,
+	"CommunityID":           communityid.New,
+	"DecodeJSONFields":      actions.NewDecodeJSONFields,
+	"Dissect":               dissect.NewProcessor,
+	"DNS":                   dns.New,
+	"Rename":                actions.NewRenameFields,
 }
```

Comment:
We might want to add theses https://github.com/elastic/beats/pull/11297 ?

Reply:
Yes, I'll add them and expose their constructors. I'm not really happy about having this list but until I come up with a better way to reuse the existing processor registry this works.

Reply:
Constructors for copy_fields and truncate_fields are now exposed.